fenix/app/src/main/java/org/mozilla/fenix/gleanplumb/MessageNotificationWorker.kt

```diff
@@ -154,13 +153,10 @@ class MessageNotificationWorker(
             val notificationConfig = featureConfig.notificationConfig
             val pollingInterval = notificationConfig.refreshInterval.toLong()
 
-            val messageWorkRequest = PeriodicWorkRequest.Builder(
-                MessageNotificationWorker::class.java,
+            val messageWorkRequest = PeriodicWorkRequestBuilder<MessageNotificationWorker>(
                 pollingInterval,
                 TimeUnit.MINUTES,
-            ) // Only start polling after the given interval.
-                .setInitialDelay(pollingInterval, TimeUnit.MINUTES)
-                .build()
+            ).build()
 
             val instanceWorkManager = WorkManager.getInstance(context)
             instanceWorkManager.enqueueUniquePeriodicWork(
```

Comment:
Using REPLACE would change the behaviour to be less deterministic. Each time the app hits onResume state we would register a new Worker, cancelling and replacing any existing Worker. This would mean if:

1. The app is opened, the Worker is registered (2 hour interval set)
2. App is in background Worker is still running (1 hour to trigger)
3. App is re-opened the existing Worker is cancelled & replaced (2 hour interval set)

So if our worker has only a few seconds left to trigger and the app state is onResume the Worker timer is reset to the 2 hours. This could have the effect of postponing the Worker from triggering for an indefinite amount of time.

My understanding was that we would want to poll at known intervals, do we want to change this?

Reply:
> So if our worker has only a few seconds left to trigger and the app state is onResume the Worker timer is reset to the 2 hours. 

My understanding of the initial delay moving to zero would be that this would be executed as soon as it was scheduled, i.e. at app startup, and every `pollingInterval` afterwards.

Reply:
That is the case for the first set-up e.g fresh install, but subsequent intervals would use our polling interval, so if the app was closed/opened our existing worker would still be registered so no immediate execution